### PR TITLE
NH IMEX improvements

### PR DESCRIPTION
**Pull request by mt5555**
1.  Switch newton iteration to iterate on w instead of PHI.   This formulation removes all scans (for improved GPU performance), improves precision (allowing residuals down to 1e-11) and makes for better normalization of the Newton increment and residual norms.  

2.  Add "tstep_type=9", an IMEX method based on the KGU53 explicit table used by the hydrostatic code, and adding a 2nd order implicit table.

3.  adjust reference profiles to those found best for HS+topo simulations (but resulting in a degradation in the DCMIP Shar mountain test case)

4. remove obsolete IMEX timestepping algorithms.  

[non-BFB] for theta-l H and NH tests

testing and documentation of KGU53-IMEX method:
https://acme-climate.atlassian.net/wiki/spaces/NGDNA/pages/1041401881/IMEX+methods

### CODE REVIEW

File: components/homme/src/theta-l/imex_mod.F90 [hunk 1]
```diff
@@ -69,31 +73,29 @@ subroutine compute_stage_value_dirk(n0,np1,alphadt,qn0,dt2,elem,hvcoord,hybrid,&
     ! compute_gwphis()            used to compute g*a(k) u(nt) dot grad_phis  in SPHI(nt)   
     !
     ! We then precompute:
-    !  w_rhs = w_0 + alphadt1 * SW(n0)  + alphadt1*SW(nm1) 
-    !  phi_rhs = w_0 + alphadt1 * SPHI(n0)  + alphadt1*SPHI(nm1) -  dt2*compute_gwphi(np1)
+    !  w_rhs = w_0 + alphadt_n0 * SW(n0)  + alphadt_nm1*SW(nm1) 
+    !  phi_rhs = w_0 + alphad_n0 * SPHI(n0)  + alphadt1_nm1*SPHI(nm1) -  dt2*compute_gwphi(np1)
```
Comment (ambrad) [inline components/homme/src/theta-l/imex_mod.F90]: spelling: iteration

File: components/homme/src/theta-l/imex_mod.F90 [hunk 2]
```diff
@@ -113,12 +114,13 @@ subroutine compute_stage_value_dirk(n0,np1,alphadt,qn0,dt2,elem,hvcoord,hybrid,&
     real (kind=real_kind) :: Jac2U(nlev-1,np,np)
 
 
-    real (kind=real_kind) :: wgdtmax
+    real (kind=real_kind) :: wmax
```
Comment (ambrad) [inline components/homme/src/theta-l/imex_mod.F90]: Switch `real*8` below to `real (kind=real_kind)`.

File: components/homme/src/theta-l/model_init_mod.F90 [hunk 3]
```diff
@@ -84,6 +84,19 @@ subroutine model_init2(elem,hybrid,deriv,hvcoord,tl,nets,nete )
       call phi_from_eos(hvcoord,elem(ie)%state%phis,&
            temp,elem(ie)%derived%dp_ref,elem(ie)%derived%phi_ref)
       elem(ie)%derived%theta_ref=0
+#endif
+#ifdef HV_REFSTATES_V2
```
Comment (ambrad) [inline components/homme/src/theta-l/model_init_mod.F90]: Does this change affect hydrostatic runs? Is this something for Luca to incorporate in his next merge?

File: components/homme/src/theta-l/prim_advance_mod.F90 [hunk 4]
```diff
@@ -118,10 +118,9 @@ subroutine prim_advance_exp(elem, deriv, hvcoord, hybrid,dt, tl,  nets, nete, co
 !                 optimal: for windspeeds ~120m/s,gravity: 340m/2
 !                 run with qsplit=1
 !                 (K&G 2nd order method has CFL=4. tiny CFL improvement not worth 2nd order)
-!   tstep_type=6  KG4+BE/CN  2nd order
 !   tstep_type=7  IMKG254a
 !   tstep_type=8  KG3+BE/CN  KG3 2nd order explicit, 1st order off-centering implicit
-!   tstep_type=9  KG5+BE/CN  KG5(3rd order linear, 3.87CFL) + BE/CN,  2nd order IMEX
+!   tstep_type=9  KGU53+BE/CN  KGU53 3rd order explicit, 1st order off-centering implicit
```
Comment (ambrad) [inline components/homme/src/theta-l/prim_advance_mod.F90]: Is this still supposed to say "1st order off-centering implicit"?

Comment (mt5555) [reply components/homme/src/theta-l/prim_advance_mod.F90]: fixed

File: components/homme/src/theta-l/model_init_mod.F90 (hunk 3 above)
Comment (mt5555) [reply components/homme/src/theta-l/model_init_mod.F90]: yes - but I think this code Luca will still be using the Fortran.

**Comment by AaronDonahue** [comment]
@rljacob and @mt5555 , I've merged this to next and I think it will need a bless on the theta tests (if I'm not mistaken).  But I'm guessing that means it would also need a bless when merged to master.  Given the holiday week should I hold off on the merge to master until the New Year?

**Comment by rljacob** [comment]
A bless is only done when merging to master.   If you can merge to master tomorrow and issue the bless request, that should be ok.

**Comment by AaronDonahue** [comment]
Ok cool.

**Comment by rljacob** [comment]
@mt5555 this also caused the HOMME and HOMMEBFB tests to fail.  Is that expected?

**Comment by AaronDonahue** [comment]
Those tests appear to also be "theta" related:
```
The following tests FAILED:
	 11 - thetah-test22 (Failed)
	 12 - thetanh-test22 (Failed)
	 14 - thetah-TC-ftype4 (Failed)
	 15 - thetah-TC (Failed)
	 16 - thetanh-TC (Failed)
	 17 - thetanh-TC-nudiv (Failed)
	 18 - thetanh-TC-nutop (Failed)
	 19 - thetanhwet-TC (Failed)
```

**Comment by mt5555** [comment]
yes, HOMMEBFB failures are also expected.

**Comment by AaronDonahue** [comment]
@mt5555 I also see a fail on Anvil for test HOMMEBFB_P24.f19_g16_rx1.A.anvil_intel.  It looks like the test fails because there are now 11 test cases instead of 5.  I'm guessing this is expected as well?

**Comment by oksanaguba** [comment]
on anvil hommebfb won't work because there is no cmake file for it. it won't build.

**Comment by AaronDonahue** [comment]
ahhh ok. Just wanted to make sure.

**Pull request merged**